src/webui/webapplication.cpp

```diff
@@ -610,28 +615,35 @@ void WebApplication::sessionStart()
     m_currentSession->registerAPIController<TransferController>(u"transfer"_qs);
     m_sessions[m_currentSession->id()] = m_currentSession;
 
-    QNetworkCookie cookie(C_SID, m_currentSession->id().toUtf8());
-    cookie.setHttpOnly(true);
-    cookie.setSecure(m_isSecureCookieEnabled && m_isHttpsEnabled);
-    cookie.setPath(u"/"_qs);
-    QByteArray cookieRawForm = cookie.toRawForm();
-    if (m_isCSRFProtectionEnabled)
-        cookieRawForm.append("; SameSite=Strict");
-    setHeader({Http::HEADER_SET_COOKIE, QString::fromLatin1(cookieRawForm)});
+    const auto sessionId = m_currentSession->id().toUtf8();
+    if (m_request.headers.contains(HEADER_API_TOKEN)) {
+        setHeader({HEADER_API_TOKEN, QString::fromLatin1(sessionId)});
```

Comment:
Why do we need to echo the token back in the response?

Reply:
Based on the PR description, this is for retrieving the initial token value. It might be worthwhile to have an explicit endpoint for this, rather than requiring the user to specify a blank auth header in the request.

Reply:
>Based on the PR description, this is for retrieving the initial token value.

IIRC, this is for asking the server to use tokens instead of cookies.

Reply:
Added new `auth/auth` endpoint for it.